vulture/lines.py

```diff
@@ -23,27 +22,18 @@
     })
 
 
-# Reinventing wheel here to keep vulture lightweight.
-def memoize(func):
-    _cache = {}
+def get_last_line_number(node):
+    last_body = node.body[-1]
```

Comment:
The fact that we use [-1] twice suggests that this can be simplified. I think we should remove the line `last_body = node.body[-1]` and rename all occurences of "last_body" to "node" in the code below. This will probably make some tests fail. However, the current code doesn't handle all cases correctly either. For example, it fails if node doesn't have a field "body". I think the following code might be closer to what we're after (untested):

```
def _get_last_child(node):
    for name in ['finalbody', 'orelse', 'handlers', 'returns', 'body']:
        try:
            field = getattr(node, name)
        except AttributeError:
            continue

        if isinstance(field, list):
            if field:
                return field[-1]
        else:
            return field
    return None


def get_last_line_number(node):
    while True:
        last_child = _get_last_child(node)
        if last_child is None:
            return node.lineno
        node = last_child
```